solar_consumer/save/save_data_platform.py

```diff
@@ -0,0 +1,173 @@
+"""Functions to save to the Data-platform
+
+https://github.com/openclimatefix/data-platform
+
+"""
+
+from loguru import logger
+from datetime import datetime, timezone
+
+from dp_sdk.ocf import dp
+import pandas as pd
+
+
+async def save_generation_to_data_platform(
+    data_df: pd.DataFrame, client: dp.DataPlatformDataServiceStub | None = None
+):
+    """
+    Save Generation data to the Data-platform.
+
+    0. First we get all the locations
+
+    Here's how we do it for each gsp
+    1. Get all the locations
+    2. Create an observer for that regime if it doesn't already exist
+    3. For each gsp: Get only the data for that gsp
+    4. Get the location for that gsp
+    5. Get the most recent observations for that location and observer,
+    6. Remove any data points from our data that are already in the database
+    7. Update location capacity based on the max capacity in this data
+    8. Create new observations for the remaining data points
+
+    :param data_df: DataFrame containing forecast data with required columns.
+    """
+
+    assert "target_datetime_utc" in data_df.columns
+    assert "solar_generation_kw" in data_df.columns
+    assert "gsp_id" in data_df.columns
+    assert "regime" in data_df.columns
+    assert "capacity_mwp" in data_df.columns
+
+    # 1. Get all locations (UK GSPs and National)
+    all_locations = await get_all_gsp_and_national_locations(client)
+
+    # 2. Create an observer for that gsp and regime if it doesn't already exist
```

Comment:
Also this says for "that" GSP - but it seems like the same observer is being used for all GSPs (and National) here - which is what I would expect, since this is all pulling from the same source